packages/eas-cli/src/metadata/apple/tasks/age-rating.ts

```diff
@@ -0,0 +1,42 @@
+import { AgeRatingDeclaration } from '@expo/apple-utils';
+import assert from 'assert';
+import chalk from 'chalk';
+
+import Log from '../../../log';
+import { logAsync } from '../../utils/log';
+import { AppleTask, TaskDownloadOptions, TaskPrepareOptions, TaskUploadOptions } from '../task';
+
+export type AgeRatingData = {
+  /** The app age rating declaration for the app version */
+  ageRating: AgeRatingDeclaration;
+};
+
+export class AgeRatingTask extends AppleTask {
+  name = (): string => 'age rating declarations';
+
+  async prepareAsync({ context }: TaskPrepareOptions): Promise<void> {
+    assert(context.version, `App version information is not prepared, can't update age rating`);
+    context.ageRating = (await context.version.getAgeRatingDeclarationAsync()) || undefined;
+  }
+
+  async downloadAsync({ config, context }: TaskDownloadOptions): Promise<void> {
```

Comment:
It does not look like this method is doing what the description in abstract class is claiming (the same applies to other tasks)

Reply:
Are you referring to this?
> Download all information from the App Store to generate the store configuration

If so, I think it does. The only thing `downloadAsync` is doing is pulling information from the stores. That would be done by loading each individual models or entities (Download all information from the App Store), and populating the `AppleConfigWriter` (to generate the store configuration).

It does not generate the store config itself, that's what the `AppleConfigWriter` is doing. But it is required to be able to generate that config.